dvc/fs/dvc.py

```diff
@@ -187,6 +187,7 @@ def _get_key(self, path: "StrPath") -> Key:
         return parts
 
     def _get_key_from_relative(self, path) -> Key:
+        path = self._strip_protocol(path)
```

Comment:
We never use the `dvc://` protocol in DVC so this wasn't an issue before, but we should still be stripping it when needed so that `api.DVCFileSystem` works properly as a generic registered fsspec implementation